### PR TITLE
Implementing Gob encoding/decoding for the classic BloomFilter

### PR DESCRIPTION
**Pull request by alexk307**

### CODE REVIEW

**Comment by alexk307** [comment]
@tylertreat Only implemented the classic bloom filter for now. I followed the same strategy put forth in the scalable bloom filter module.

File: classic.go [hunk 1]
```diff
@@ -119,3 +122,72 @@ func (b *BloomFilter) Reset() *BloomFilter {
...
+	}
+	err = binary.Write(stream, binary.BigEndian, uint64(b.k))
+	if err != nil {
+		return 0, err
+	}
+	if _, err := b.buckets.WriteTo(stream); err != nil {
+		return 0, err
+	}
+
+	return int64(5 * binary.Size(uint64(0))), err
```
Comment (tylertreat) [inline classic.go]: This isn't right? It will always return 5.

File: classic.go [hunk 2]
```diff
@@ -119,3 +122,72 @@ func (b *BloomFilter) Reset() *BloomFilter {
...
+	if err != nil {
+		return 0, err
+	}
+
+	buckets.ReadFrom(stream)
+	b.count = uint(count)
+	b.m = uint(m)
+	b.k = uint(k)
+	b.buckets = &buckets
+	return int64(5 * binary.Size(uint64(0))), nil
```
Comment (tylertreat) [inline classic.go]: ditto

Comment (tylertreat) [inline classic.go]: ditto

File: classic.go (hunk 1 above)
Comment (alexk307) [inline classic.go]: Ah, meant to fix this before pushing. Fixed.

**Comment by tylertreat** [comment]
Thanks!

**Comment by alexk307** [comment]
@tylertreat thanks for the quick review/merge! I should be implementing encoding/decoding for at least the inverse filter (as that's what my use case needs).

**Pull request merged**